whatlies/embeddingset.py

```diff
@@ -549,6 +568,81 @@ def plot_correlation(self, metric=None):
         plt.setp(ax.get_xticklabels(), rotation=90, ha="right", rotation_mode="anchor")
         plt.show()
 
+    def plot_similarity(self, metric="cosine", norm=True):
+        """
+        Make a similarity plot. Shows you the similarity between all the word embeddings in the set.
+
+        Arguments:
+            metric: cosine or euclidean
+
+        Usage:
+
+        ```python
+        from whatlies.language import SpacyLanguage
+        lang = SpacyLanguage("en_core_web_md")
+
+        names = ['red', 'blue', 'green', 'yellow', 'cat', 'dog', 'mouse', 'rat', 'bike', 'car']
+        emb = lang[names]
+        emb.plot_distance()
+        ```
+
+        ![](https://rasahq.github.io/whatlies/images/corrplot.png)
+        """
+        df = self.to_dataframe().T
```

Comment:
I was wondering what's the point of using `to_dataframe` here. As it seems, you are just using the `columns` attribute of this dataframe which is basically equivalent to `list(self.embeddings.keys())`!

Reply:
It's an artifact left over from the previous implementation. Originally I also wanted to have `correlation` as a metric which is super easy to calculate via `df.corr()`. The code definately is work in progress. I certianly think a cleanup is possible here.